### PR TITLE
JE-239: Split billing into two functions. Grouped marketing account i…

### PR DESCRIPTION
**Pull request by tuj**
* Split invoice generation into two functions since they differ if they are (not?) marketing account orders.
* Grouped invoices by ilbrary when marketing account is selected.
* Updated bundles.

### CODE REVIEW

File: bundles/Billing/Controller/IndexController.php [hunk 1]
```diff
@@ -21,7 +21,6 @@ class IndexController extends AbstractController
     /**
      * @Route("/show_export_invoice/{invoiceId}", name="api_show_export_invoice", methods={"GET"})
      *
-     * @param \Billing\Service\BillingService $billingService
      * @param $invoiceId
```
Comment (rimi-itk) [inline bundles/Billing/Controller/IndexController.php:24]: Is this part of our new documentation strategy, i.e. to only “document” select parameters?

File: bundles/GraphicServiceBilling/Controller/MainController.php [hunk 2]
```diff
@@ -32,15 +32,13 @@ class MainController extends AbstractController
...
-     * @param \Symfony\Component\HttpFoundation\Request                   $request
-     * @param \App\Service\MenuService                                    $menuService
-     * @param \GraphicServiceBilling\Service\GraphicServiceBillingService $graphicServiceBillingService
      * @param $boundProjectId
      *
      * @return \Symfony\Component\HttpFoundation\Response
      *
      * @throws \PhpOffice\PhpSpreadsheet\Exception
      * @throws \PhpOffice\PhpSpreadsheet\Writer\Exception
+     * @throws \Exception
```
Comment (rimi-itk) [inline bundles/GraphicServiceBilling/Controller/MainController.php]: Can we make this a more specific exception?

File: bundles/GraphicServiceBilling/Service/GraphicServiceBillingService.php [hunk 3]
```diff
@@ -54,30 +54,141 @@ public function __construct(
...
+
+            $library = $task->fields->{$libraryFieldId};
+
+            if (isset($entries[$library])) {
+                $header = $entries[$library]->header;
+            } else {
+                // Create header line data.
+                $header = (object) [
+                    'debtor' => $this->boundReceiverAccount,
+                    'salesChannel' => '10',
```
Comment (rimi-itk) [inline bundles/GraphicServiceBilling/Service/GraphicServiceBillingService.php:83]: Should this (`salesChannel`) be configurable?

File: bundles/GraphicServiceBilling/Service/GraphicServiceBillingService.php [hunk 4]
```diff
@@ -54,30 +54,141 @@ public function __construct(
...
+        $orderLinesFieldId = $this->billingService->getCustomFieldId('Order lines');
+        $entries = [];
+
+        foreach ($tasks as $task) {
             $description = implode('', [
-                $task->fields->summary,
-                ' ('.$task->key.'): ',
-                $description,
+                $task->key.': ',
+                preg_replace('/\\\\\\\\/', '.', $task->fields->{$orderLinesFieldId}),
```
Comment (rimi-itk) [inline bundles/GraphicServiceBilling/Service/GraphicServiceBillingService.php]: A comment about what's going on here would be nice. Consider adding a `getTaskDescription` method to handle this.

**Review by rimi-itk — CHANGES_REQUESTED** [review]

File: bundles/Billing/Controller/IndexController.php (hunk 1 above)
Comment (tuj) [reply bundles/Billing/Controller/IndexController.php:24]: This is the result of applying cs-fixer where the new rule "no_superfluous_phpdoc_tags" has been added.

File: bundles/GraphicServiceBilling/Service/GraphicServiceBillingService.php (hunk 3 above)
Comment (tuj) [reply bundles/GraphicServiceBilling/Service/GraphicServiceBillingService.php:83]: Not in this case.

File: bundles/GraphicServiceBilling/Service/GraphicServiceBillingService.php (hunk 4 above)
Comment (tuj) [reply bundles/GraphicServiceBilling/Service/GraphicServiceBillingService.php]: Yes.

File: bundles/GraphicServiceBilling/Controller/MainController.php (hunk 2 above)
Comment (tuj) [reply bundles/GraphicServiceBilling/Controller/MainController.php]: Cleaned up

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**